packages/smooth_app/lib/helpers/product_compatibility_helper.dart

```diff
@@ -0,0 +1,100 @@
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:openfoodfacts/model/Attribute.dart';
+import 'package:openfoodfacts/model/AttributeGroup.dart';
+import 'package:openfoodfacts/model/Product.dart';
+import 'package:openfoodfacts/personalized_search/preference_importance.dart';
+import 'package:smooth_app/data_models/product_preferences.dart';
+import 'package:smooth_app/helpers/attributes_card_helper.dart';
+
+enum ProductCompatibility {
+  UNKNOWN,
+  INCOMPATIBLE,
+  BAD_COMPATIBILITY,
+  NEUTRAL_COMPATIBILITY,
+  GOOD_COMPATIBILITY,
+}
+
+const int _BAD_COMPATIBILITY_UPPER_THRESHOLD = 33;
+const int _NEUTRAL_COMPATIBILITY_UPPER_THRESHOLD = 66;
+
+// Defines the weight of an attribute while computing the average match score
+// for the product. The weight depends upon it's importance set in user prefs.
+const Map<String, int> attributeImportanceWeight = <String, int>{
+  PreferenceImportance.ID_MANDATORY: 4,
+  PreferenceImportance.ID_VERY_IMPORTANT: 2,
+  PreferenceImportance.ID_IMPORTANT: 1,
+  PreferenceImportance.ID_NOT_IMPORTANT: 0,
+};
+
+Color getProductCompatibilityHeaderBackgroundColor(
+    ProductCompatibility compatibility) {
+  switch (compatibility) {
+    case ProductCompatibility.UNKNOWN:
+      return Colors.grey;
+    case ProductCompatibility.INCOMPATIBLE:
+      return Colors.red;
+    case ProductCompatibility.BAD_COMPATIBILITY:
+      return Colors.orangeAccent;
+    case ProductCompatibility.NEUTRAL_COMPATIBILITY:
+      return Colors.amber;
+    case ProductCompatibility.GOOD_COMPATIBILITY:
+      return Colors.green;
+  }
+}
+
+String getProductCompatibilityHeaderTextWidget(
+  ProductCompatibility compatibility,
+) {
+  // Note: This text should come from BE.
```

Comment:
The compatibility can't be evaluated from the backend though, so we'll probably need to keep these strings in the client. They should be internationalized though.

Reply:
Makes sense, I'll get to setting up internationalization for this (and maybe other strings) later. Added a TODO